DQM/L1TMonitor/src/L1TObjectsTiming.cc

```diff
@@ -0,0 +1,480 @@
+      stage2CaloLayer2TauToken_(consumes<l1t::TauBxCollection>(ps.getParameter<edm::InputTag>("stage2CaloLayer2TauProducer"))),
+      stage2CaloLayer2EtSumToken_(consumes<l1t::EtSumBxCollection>(ps.getParameter<edm::InputTag>("stage2CaloLayer2EtSumProducer"))),
+      l1tStage2uGtProducer_(consumes<GlobalAlgBlkBxCollection>(ps.getParameter<edm::InputTag>("ugtProducer"))),
+      monitorDir(ps.getUntrackedParameter<std::string>("monitorDir")),
+      verbose(ps.getUntrackedParameter<bool>("verbose")),
+      algoBitFirstBxInTrain_(-1),
+      algoBitLastBxInTrain_(-1),
+      algoNameFirstBxInTrain_(ps.getUntrackedParameter<std::string>("firstBXInTrainAlgo","")),
+      algoNameLastBxInTrain_(ps.getUntrackedParameter<std::string>("lastBXInTrainAlgo",""))
+{
+    // Get the trigger bits corresponding to the trigger names from the event setup.
+    // Use the same input tag for everything
+    const auto algInputTag = ps.getParameter<edm::InputTag>("ugtProducer");
+    const auto extInputTag = ps.getParameter<edm::InputTag>("ugtProducer");
+    gtUtil_ = new l1t::L1TGlobalUtil(ps, consumesCollector(), *this, algInputTag, extInputTag);
+
+//    const unsigned int bxrange = 5; //this is the out bx range 
+    muons_eta_phi.reserve(bxrange);
+    jet_eta_phi.reserve(bxrange);
+    egamma_eta_phi.reserve(bxrange);
+    tau_eta_phi.reserve(bxrange);
+    etsum_eta_phi_MET.reserve(bxrange);
+    etsum_eta_phi_METHF.reserve(bxrange);
+    etsum_eta_phi_MHT.reserve(bxrange);
+    etsum_eta_phi_MHTHF.reserve(bxrange);
+    muons_eta_phi_isolated.reserve(bxrange);
+    jet_eta_phi_isolated.reserve(bxrange);
+    egamma_eta_phi_isolated.reserve(bxrange);
+    tau_eta_phi_isolated.reserve(bxrange);
+    etsum_eta_phi_MET_isolated.reserve(bxrange);
+    etsum_eta_phi_METHF_isolated.reserve(bxrange);
+    etsum_eta_phi_MHT_isolated.reserve(bxrange);
+    etsum_eta_phi_MHTHF_isolated.reserve(bxrange);
+    muons_eta_phi_firstbunch.reserve(bxrange);
+    jet_eta_phi_firstbunch.reserve(bxrange);
+    egamma_eta_phi_firstbunch.reserve(bxrange);
+    tau_eta_phi_firstbunch.reserve(bxrange);
+    etsum_eta_phi_MET_firstbunch.reserve(bxrange);
+    etsum_eta_phi_METHF_firstbunch.reserve(bxrange);
+    etsum_eta_phi_MHT_firstbunch.reserve(bxrange);
+    etsum_eta_phi_MHTHF_firstbunch.reserve(bxrange);
+    muons_eta_phi_lastbunch.reserve(bxrange);
+    jet_eta_phi_lastbunch.reserve(bxrange);
+    egamma_eta_phi_lastbunch.reserve(bxrange);
+    tau_eta_phi_lastbunch.reserve(bxrange);
+    etsum_eta_phi_MET_lastbunch.reserve(bxrange);
+    etsum_eta_phi_METHF_lastbunch.reserve(bxrange);
+    etsum_eta_phi_MHT_lastbunch.reserve(bxrange);
+    etsum_eta_phi_MHTHF_lastbunch.reserve(bxrange);
+    
+}
+
+L1TObjectsTiming::~L1TObjectsTiming() {}
+
+void L1TObjectsTiming::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("muonProducer")->setComment("uGMT output muons.");;
```

Comment:
Better use these as descriptions for the inputs:
L1T muons
L1T jets
L1T egamma
L1T taus
L1T etsums